File: sandbox/plc4go/internal/plc4go/modbus/readwrite/ModbusPDUGetComEventLogRequest.go
//
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
//
package readwrite

import (
	"plc4x.apache.org/plc4go-modbus-driver/0.8.0/internal/plc4go/spi"
)

// The data-structure of this message
type ModbusPDUGetComEventLogRequest struct {
	ModbusPDU
}

// The corresponding interface
type IModbusPDUGetComEventLogRequest interface {
	IModbusPDU
	Serialize(io spi.WriteBuffer)
}

// Accessors for discriminator values.
func (m ModbusPDUGetComEventLogRequest) ErrorFlag() bool {
	return false
}

func (m ModbusPDUGetComEventLogRequest) FunctionFlag() uint8 {
	return 0x0C
}

func (m ModbusPDUGetComEventLogRequest) Response() bool {
	return false
}

func (m ModbusPDUGetComEventLogRequest) initialize() spi.Message {
	return spi.Message(m)
}

func NewModbusPDUGetComEventLogRequest() ModbusPDUInitializer {
	return &ModbusPDUGetComEventLogRequest{}
}

func (m ModbusPDUGetComEventLogRequest) LengthInBits() uint16 {
	var lengthInBits uint16 = m.ModbusPDU.LengthInBits()

	return lengthInBits
}

func (m ModbusPDUGetComEventLogRequest) LengthInBytes() uint16 {
	return m.LengthInBits() / 8
}

func ModbusPDUGetComEventLogRequestParse(io spi.ReadBuffer) (ModbusPDUInitializer, error) {

	// Create the instance
	return NewModbusPDUGetComEventLogRequest(), nil
}

func (m ModbusPDUGetComEventLogRequest) Serialize(io spi.WriteBuffer) {
	serializeFunc := func(typ interface{}) {
		if _, ok := typ.(IModbusPDU); ok {
		}
	}
	serializeFunc(m)
}

File: plc4py/plc4py/protocols/modbus/readwrite/ModbusPDUGetComEventLogRequest.py
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#

from dataclasses import dataclass

from plc4py.api.exceptions.exceptions import PlcRuntimeException
from plc4py.api.exceptions.exceptions import SerializationException
from plc4py.api.messages.PlcMessage import PlcMessage
from plc4py.protocols.modbus.readwrite.ModbusPDU import ModbusPDU
from plc4py.spi.generation.ReadBuffer import ReadBuffer
from plc4py.spi.generation.WriteBuffer import WriteBuffer
from plc4py.utils.ConnectionStringHandling import strtobool
from typing import ClassVar
import math


@dataclass
class ModbusPDUGetComEventLogRequest(ModbusPDU):
    # Accessors for discriminator values.
    error_flag: ClassVar[bool] = False
    function_flag: ClassVar[int] = 0x0C
    response: ClassVar[bool] = False

    def serialize_modbus_pdu_child(self, write_buffer: WriteBuffer):
        write_buffer.push_context("ModbusPDUGetComEventLogRequest")

        write_buffer.pop_context("ModbusPDUGetComEventLogRequest")

    def length_in_bytes(self) -> int:
        return int(math.ceil(float(self.length_in_bits() / 8.0)))

    def length_in_bits(self) -> int:
        length_in_bits: int = super().length_in_bits()
        _value: ModbusPDUGetComEventLogRequest = self

        return length_in_bits

    @staticmethod
    def static_parse_builder(read_buffer: ReadBuffer, response: bool):
        read_buffer.push_context("ModbusPDUGetComEventLogRequest")

        if isinstance(response, str):
            response = bool(strtobool(response))

        read_buffer.pop_context("ModbusPDUGetComEventLogRequest")
        # Create the instance
        return ModbusPDUGetComEventLogRequestBuilder()

    def equals(self, o: object) -> bool:
        if self == o:
            return True

        if not isinstance(o, ModbusPDUGetComEventLogRequest):
            return False

        that: ModbusPDUGetComEventLogRequest = ModbusPDUGetComEventLogRequest(o)
        return super().equals(that) and True

    def hash_code(self) -> int:
        return hash(self)

    def __str__(self) -> str:
        # TODO:- Implement a generic python object to probably json convertor or something.
        return ""


@dataclass
class ModbusPDUGetComEventLogRequestBuilder:

    def build(
        self,
    ) -> ModbusPDUGetComEventLogRequest:
        modbus_pduget_com_event_log_request: ModbusPDUGetComEventLogRequest = (
            ModbusPDUGetComEventLogRequest()
        )
        return modbus_pduget_com_event_log_request
